### DISCUSSION TITLE
How to get password from Peers Config File?

### DISCUSSION THREAD

**Discussion by KeyErrorFinn** [rustdesk/rustdesk#3179] [Q&A]
I am wondering how to get the passwords from the peers config folder. currently it lays the password out in a list with numbers but I don't know how to decrypt it. How would I pull the password from it? is it encrypted?

**Comment by muelli** [comment]
The password you type in is [hashed](https://github.com/rustdesk/rustdesk/blob/77064cc2f8c359c0af69a51c5828826c626f802b/src/client.rs#L3459-L3465) [twice](https://github.com/rustdesk/rustdesk/blob/77064cc2f8c359c0af69a51c5828826c626f802b/src/client.rs#L3467-L3471) and [encrypted](https://github.com/rustdesk/hbb_common/blob/32fed54062c1cdf18146899515ed2850f6ff986b/src/config.rs#L1334-L1335) before it's stored. The encryption you can probably remove but then you still have the doubly hashed (and salted) string.

The answer, in short: You cannot recover the password you typed in from the stored configurations.
